backend/webapp/src/main/java/org/bonitasoft/web/designer/controller/UploadAssetController.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.web.designer.controller;
+
+import javax.inject.Inject;
+
+import org.bonitasoft.web.designer.controller.upload.AssetUploader;
+import org.bonitasoft.web.designer.model.page.Page;
+import org.bonitasoft.web.designer.rest.ErrorMessage;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * This controller is used to upload resources and attach them to a component
+ */
+@Controller
+public class UploadAssetController {
```

Comment:
By the way, we should move rest package inside controller one maybe ...
Might be done later